presto-verifier/src/main/java/com/facebook/presto/verifier/framework/VerificationManager.java

```diff
@@ -330,4 +371,97 @@ private void postEvent(VerifierQueryEvent event)
             eventClient.post(event);
         }
     }
+
+    private class LimitWithoutOrderByChecker
+            extends AstVisitor<Boolean, Integer>
+    {
+        @Override
+        protected Boolean visitQuerySpecification(QuerySpecification node, Integer level)
+        {
+            //Check if node is not at root level and limit clause is present without order by
+            return level != 0 && node.getLimit().isPresent() && !node.getOrderBy().isPresent();
+        }
+
+        @Override
+        protected Boolean visitQuery(Query node, Integer level)
+        {
+            if (node.getWith().isPresent()) {
+                With with = node.getWith().get();
+                level++;
+                for (WithQuery query : with.getQueries()) {
+                    if (process(query.getQuery(), level)) {
+                        return true;
+                    }
+                }
+                level--;
```

Comment:
Don't modify param. Just do level + 1 when you call recursively.

Reply:
done